File: app/src/main/java/pt/isec/supraindustries/tp_amov/oldfiles/Lista.java
package pt.isec.supraindustries.tp_amov.oldfiles;

import java.util.List;

import pt.isec.supraindustries.tp_amov.Data.Produto;

public class Lista {
    String nome;
    List<pt.isec.supraindustries.tp_amov.Data.Produto> produtoList;

    public Lista(String nome) {
        this.nome = nome;
    }

    public void addProduto(Produto produto){
        produtoList.add(produto);
    }
}

File: app/src/main/java/pt/isec/supraindustries/tp_amov/Data/Lista.kt
package pt.isec.supraindustries.tp_amov.Data

import java.io.Serializable

data class Lista(var nome: String) : Serializable {
    var lista : HashMap<Produto, Int> = HashMap<Produto, Int>()

    fun addProduto(produto: Produto, quantidade: Int){
        lista.set(produto, quantidade)
    }

    fun removeProduto(produto: Produto){
        lista.remove(produto)
    }

    fun setName(name : String){
        nome = name
    }
}